blockmanager.go

```diff
@@ -1072,6 +1141,40 @@ func (b *blockManager) handleInvMsg(imsg *invMsg) {
 	}
 }
 
+// limitMap is a helper function for maps that require a maximum limit by
+// evicting a random rejected transaction if adding a new value would cause it
+// to overflow the maximum allowed.
+func (b *blockManager) limitMap(m map[wire.ShaHash]struct{}, limit int) error {
+	if len(m)+1 > limit {
+		// Generate a cryptographically random hash.
+		randHashBytes := make([]byte, wire.HashSize)
+		_, err := rand.Read(randHashBytes)
```

Comment:
So, I'm not sure we really need strong _cryptographic_ randomness here. I suspect reading from `/dev/urandom` may become an unnecessary bottleneck when processing transactions. In order to speed up the execution of this function, we may want to instead use a fast PRNG such as [xorshift](https://en.wikipedia.org/wiki/Xorshift). There exists a pure-go implementation, which can be found [here](https://github.com/lazybeaver/xorshift). 

NOTE: I haven't benchmarked this yet, but this is a recommendation based on past experience when optimizing the `SigCache` in #598.